Source/ACE.Server/Factories/LootGenerationFactory_Missile.cs

```diff
@@ -29,63 +33,68 @@ public static WorldObject CreateMissileWeapon(TreasureDeath profile, bool isMagi
+            wo.LongDesc = wo.Name;
 
-            // MagicD/Missile Bonus
+            // MeleeD/MagicD/Missile Bonus
             wo.WeaponMagicDefense = GetMagicMissileDMod(profile.Tier); 
             wo.WeaponMissileDefense = GetMagicMissileDMod(profile.Tier);
+            double meleeDMod = GetWieldReqMeleeDMod(wieldDifficulty);
+            if (meleeDMod > 0.0f)
+                wo.WeaponDefense = meleeDMod;
 
-            wo.SetProperty(PropertyFloat.DamageMod, GetMissileDamageMod(wieldDifficulty, wo.GetProperty(PropertyInt.WeaponType)));
-
+            // Damage
+            wo.DamageMod = GetMissileDamageMod(wieldDifficulty, wo.GetProperty(PropertyInt.WeaponType));
             if (elemenatalBonus > 0)
-                wo.SetProperty(PropertyInt.ElementalDamageBonus, elemenatalBonus);
+                wo.ElementalDamageBonus = elemenatalBonus;
 
+            // Wields
             if (wieldDifficulty > 0)
             {
-                wo.SetProperty(PropertyInt.WieldDifficulty, wieldDifficulty);
-                wo.SetProperty(PropertyInt.WieldRequirements, (int)WieldRequirement.RawSkill);
-                wo.SetProperty(PropertyInt.WieldSkillType, (int)Skill.MissileWeapons);
+                wo.WieldDifficulty = wieldDifficulty;
+                wo.WieldRequirements = WieldRequirement.RawSkill;
+                wo.WieldSkillType = (int)Skill.MissileWeapons;
             }
             else
             {
-                wo.RemoveProperty(PropertyInt.WieldDifficulty);
-                wo.RemoveProperty(PropertyInt.WieldRequirements);
-                wo.RemoveProperty(PropertyInt.WieldSkillType);
+                wo.WieldDifficulty = null;
+                wo.WieldRequirements = WieldRequirement.Invalid;
+                wo.WieldSkillType = null;
             }
 
+            // Magic
             if (isMagical)
                 wo = AssignMagic(wo, profile);
             else
             {
-                wo.RemoveProperty(PropertyInt.ItemManaCost);
-                wo.RemoveProperty(PropertyInt.ItemMaxMana);
-                wo.RemoveProperty(PropertyInt.ItemCurMana);
-                wo.RemoveProperty(PropertyInt.ItemSpellcraft);
-                wo.RemoveProperty(PropertyInt.ItemDifficulty);
-                wo.RemoveProperty(PropertyFloat.ManaRate);
+                wo.ItemManaCost = null;
+                wo.ItemMaxMana = null;
+                wo.ItemCurMana = null;
+                wo.ItemSpellcraft = null;
+                wo.ItemDifficulty = null;
+                wo.ManaRate = null;
             }
 
+            // Material/Value/Color
             double materialMod = LootTables.getMaterialValueModifier(wo);
             double gemMaterialMod = LootTables.getGemMaterialValueModifier(wo);
-            var value = GetValue(profile.Tier, workmanship, gemMaterialMod, materialMod);
+            var value = GetValue(profile.Tier, GetWorkmanship(profile.Tier), gemMaterialMod, materialMod);
```

Comment:
this should be using wo.Workmanship.Value

If it calls GetWorkmanship(tier) again, it will be doing a separate rng roll / getting a possibly different workmanship value than what was previously rolled for the item

Reply:
Good Catch, Fixed.